### ISSUE TITLE
Dell Latitude 5500

### ISSUE THREAD

**Issue by Dev727** [blankmac/AlpsHID#4]
I found this repo while looking for a solution that seems common on Dell I2C devices. I tried this repo and it seems to function the same as VoodooI2CHID (working TrackPad, but only one button).

I was just curious if this kext would be appropriate for my device.

**Comment by juico** [comment]
[AlpsHID_5500.zip](https://github.com/blankmac/AlpsHID/files/7359646/AlpsHID_5500.zip)
You can try this, seems like your touchpad uses the Cirque vendorID instead of the Alps. Should be the same kind of touchpad, so by adding the VendorID to the kext i think it could work.

**Comment by Dev727** [comment]
That mostly did the trick. The only thing not working is cross-communication between the Trackpoint/Pad and their buttons. Essentially the Trackpoint buttons don't work with the Trackpad and vice versa.

This isn't too shocking as I've experienced the same with VoodooPS2 Alps on Dell.

Still, this is vastly better, so thank you so much.

**Comment by Dev727** [comment]
Yeah, cross communication works on both Windows and Linux. I've worked on Trackpoint/pad kexts before, so I may be able to work on it in my limited free time, but that is correct. 2 signals get sent essentially.

I'm sure I2C is far more complex than PS2 though, so no guarantee I would be able to get it working.

**Comment by DiogoRafaelPR** [comment]
I tested the attached file "AlpsHID_5500" and I can confirm its working on Dell 5500.
The trackpad works very well, the only gesture that I tested but it didn't work, it was the smart zoom (touching twice with two fingers), but i'm not 100% sure if its a bug or my problem.

The lower physical buttons work, both left and right.
But the upper physical buttons, left and center, don't work while the right is ok.
The small mouse on the middle of the keyboard (sorry don't know the name), also works.

Also, do you think it's possible to remove the delay when dragging with one finger? That option is found on Accessibility. Both options with drag lock or without drag lock, it gives me a 1 second delay after I dragged something.
